### PR TITLE
added destructors

### PR DESCRIPTION
**Pull request by msupernaw**
Thanks for opening a pull request (PR) to FIMS! Please describe the changes under the headings below. Please also go through the checklist below to ensure the appropriate steps have been taken during development. For smaller changes, feel free to skip steps that aren't relevant, but at least describe the feature under the "what is the feature".

# What is the feature?
A brief title of this change.

# How have you implemented the solution?
Describe the changes.

# Does it impact any other area of the project?
Please describe which features are impacted by this change.

# Does this change impact the model input or output?*
Please describe any way this change impacts the model input or output.

# How to test this change
Please include a test file and/or Github workflow.

# Developer pre-PR Checklist

Please do these steps locally for big changes. Note GitHub Actions does all of these checks automatically when pushing to the repository. Please see [code development section in the contributor guide](https://noaa-fims.github.io/collaborative_workflow/contributor-guidelines.html#code-development) for details.

- [ ] Run [cmake build and ctest locally](https://noaa-fims.github.io/collaborative_workflow/testing.html#c-unit-testing-and-benchmarking) and make sure the C++ tests pass
- [ ] Run `devtools::document()` locally and push changes to the remote feature branch
- [ ] Run `styler::style_pkg()` locally and push changes to remote feature branch. If there are many changes, please do this in a separate commit.
- [ ] Run `devtools::check()` locally and make sure the package can be compiled and R tests pass. If there are failing tests, run `devtools::test(filter = "file_name")` (where "test-file_name.R" is the testthat file containing failing tests) and edit code/tests to troubleshoot tests.
- [ ] Before opening the PR, make sure all the github actions are passing on the remote feature branch.
- [ ] Make sure this PR has an informative title.

### CODE REVIEW

**Comment by k-doering-NOAA** [comment]
Thanks @msupernaw ! Sorry I didn't clearly communicate that this could be checked on later, but it would be great to add these into the code. Thanks for working on this so quickly! 

I think some accidental .DS_store files crept into this request?  I can force push a new commit to get rid of them from the git history completely and maybe add them to .gitignore so it doesn't happen again. Let me know if you would like me to do this.

@Bai-Li-NOAA , I think you brought up the destructors - does this PR look good to you? I think it does to me, but I don't really understand destructors :)

File: inst/include/common/model.hpp [hunk 1]
```diff
@@ -34,6 +34,10 @@ class Model {
   static std::shared_ptr<Model<T> > fims_model;
   std::vector<std::shared_ptr<Population<T> > > populations;
   std::vector<std::shared_ptr<Fleet<T> > > fleets;
+
```
Comment (Bai-Li-NOAA) [inline inst/include/common/model.hpp:37]: Do we need a constructor above the destructor? For example, `model(){}`

File: inst/include/common/model_object.hpp [hunk 2]
```diff
@@ -30,6 +30,9 @@ struct FIMSObject {
   std::vector<T*>
       fixed_effects_parameters; /*!< list of fixed effects parameters */
 
+    virtual ~FIMSObject(){
```
Comment (Bai-Li-NOAA) [inline inst/include/common/model_object.hpp:33]: Do we need a destructor if we are not allocating any memory here?

**Review by Bai-Li-NOAA — CHANGES_REQUESTED** [review]
Thanks @msupernaw. The changes look good to me. I do have a question about the destructor. Do we need a destructor if we are not allocating any memory in the model class and the FIMSObject struct? Or do we need to add a matching constructor for the model class and the FIMSObject struct?

@k-doering-NOAA, I agree. Getting rid of the .DS_store files and adding them to .gitignore before merging is a great idea.

**Comment by k-doering-NOAA** [comment]
I squashed and force pushed to rm the .DS_Store files entirely from the git history, and added .DS_Store to the gitignore. Matthew, note this will make your local out of sync with the remote, so you may need to [reset it](https://stackoverflow.com/questions/1628088/reset-local-repository-branch-to-be-just-like-remote-repository-head)

**Review by Bai-Li-NOAA — APPROVED** [review]

**Review by Bai-Li-NOAA — COMMENTED** [review]
The changes look good!

**Pull request merged**